test/powershell/Language/Scripting/Array.Tests.ps1

```diff
@@ -0,0 +1,54 @@
+Describe "ArrayExpression Tests" -Tags "CI" {
+    It "@([object[]](1,2,3)) should return a 3-element array of object[]" {
+        $result = @([object[]](1,2,3))
+        $result.GetType().FullName | Should Be "System.Object[]"
+        $result.Length | Should Be 3
```

Comment:
To use `Should BeOfType` I need to wrap `$result` in a new array because it will be unraveled otherwise. I feel it's clearer to just use `GetType().FullName` in this case.

As for the bad pattern, I'm still confused. Why do you think putting `$result.Length | Should Be 3` after the type check would be a bad pattern? More technically speaking, the length check should happen after we know `$result` is an array, otherwise we might (actually impossible) end up checking the `Length` property of some random type :)